std-bits/table/src/main/java/org/enso/table/data/column/operation/unary/UnaryRoundOperation.java

```diff
@@ -90,7 +84,7 @@ private void applyDouble(
       String msg = "Value is " + d;
       problemAggregator.reportArithmeticError(msg, builder.getCurrentSize());
       builder.appendNulls(1);
-    } else if (d > USE_DOUBLE_LIMIT_POSITIVE || d < USE_DOUBLE_LIMIT_NEGATIVE) {
+    } else if (!Value.asValue(d).fitsInLong()) {
```

Comment:
Reusing Truffle implementation via polyglot API. Clever. Is it fast? (enough)?

Reply:
Good point, I will take a look at the benchmarks.

Reply:
Yeah so it is tragically making things worse - ~100x slower.
![image](https://github.com/user-attachments/assets/ff83d861-9317-4790-b4f7-20ab9066ae67)

Reply:
I imagine the slowness was because we were actually falling back to `BigDecimal` every time, because `fitsInLong` was answering `false` for numbers like `2.5`. `fitsInLong` was not the right method.

I adapted the method and added a note explaining how it differs. Now running benchmarks one more time to verify all is good.